common/include/window.h

```diff
@@ -50,6 +50,79 @@ static inline void set_embedded_window_pointer(embed_window_pointer_t *wp, windo
 
 static inline void set_embedded_window_pointer(ignore_window_pointer_t *, window *) {}
 
+struct window
+{
+private:
+	grs_canvas w_canv;					// the window's canvas to draw to
+	window_event_result (*w_callback)(window *wind,const d_event &event, void *data);	// the event handler
+	int w_visible;						// whether it's visible
+	int w_modal;						// modal = accept all user input exclusively
+	void *data;							// whatever the user wants (eg menu data for 'newmenu' menus)
+	struct window *prev;				// the previous window in the doubly linked list
+	struct window *next;				// the next window in the doubly linked list
+	
+public:
+	// Declaring as friends to keep function syntax, for historical reasons (for now at least)
+	friend window *window_create(grs_canvas *src, int x, int y, int w, int h, window_subfunction<void> event_callback, void *userdata, const void *createdata);
+	
+	friend int window_close(window *wind);
+	friend int window_exists(window *wind);
+	friend window *window_get_front();
+	friend window *window_get_first();
+	friend void window_select(window &wind);
+	friend window *window_set_visible(window &wind, int visible);
+#if !DXX_USE_OGL
+	friend void window_update_canvases();
+#endif
+	friend window_event_result window_send_event(window &wind,const d_event &event);
+	friend void window_set_modal(window &wind, int modal);
+	friend int window_is_modal(window &wind);
+	
+	friend grs_canvas &window_get_canvas(window &wind)
+	{
+		return wind.w_canv;
+	}
+	
+	friend window *window_set_visible(window *wind, int visible)
+	{
+		return window_set_visible(*wind, visible);
+	}
+
+	friend int window_is_visible(window *wind)
```

Comment:
OK, yes I googled the reference form and I see it has the benefit (where intended) to disallow nullptr. However, where it uses the pointer form even chaining to the reference form would simply cause it to attempt to dereference nullptr where it has been passed to the pointer form (correct me if I'm wrong). Rather than changing all the uses of the pointer form later on, I would like to declare this and the other related functions as class methods. Therefore, I'm proposing to declare those class methods here.